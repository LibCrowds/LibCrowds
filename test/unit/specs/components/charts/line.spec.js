import BootstrapVue from 'bootstrap-vue'
import VueChartist from 'vue-chartist'
import 'chartist-plugin-tooltips'

import { mount, createLocalVue } from 'vue-test-utils'
import LineChart from '~/components/charts/Line'

describe('Line chart', () => {
  let localVue = null
  let wrapper = null

  beforeEach(() => {
    localVue = createLocalVue()
    localVue.use(BootstrapVue)
    localVue.use(VueChartist)
    wrapper = mount(LineChart, {
      localVue,
      propsData: {
        header: 'My Line Chart',
        chartData: {
          labels: ['A', 'B', 'C'],
          series: [[1, 3, 2], [4, 6, 5]]
        },
        unit: 'task'
      }
    })
  })

  it('renders correctly', () => {
    const renderer = require('vue-server-renderer').createRenderer()
    renderer.renderToString(wrapper.vm, (err, str) => {
      expect(str).toMatchSnapshot()
    })
  })
})
