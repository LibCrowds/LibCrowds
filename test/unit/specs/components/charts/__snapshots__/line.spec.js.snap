// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Line chart renders correctly 1`] = `

<div data-server-rendered="true"
     class="card"
>
  <div class="card-header">
    <div>
      My Line Chart
    </div>
  </div>
  <div class="card-body">
    <vue-chartist type="Line"
                  data="[object Object]"
                  options="[object Object]"
    >
    </vue-chartist>
  </div>
</div>

`;
